tests/test_spectral_profile.py

```diff
@@ -0,0 +1,256 @@
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pytest
+import numpy as np
+from typing import Optional, List, Union
+
+from . import logger
+
+from nomad.units import ureg
+
+from nomad_simulations import Simulation
+from nomad_simulations.model_system import ModelSystem, AtomicCell
+from nomad_simulations.atoms_state import AtomsState
+from nomad_simulations.outputs import Outputs
+from nomad_simulations.properties import (
+    SpectralProfile,
+    ElectronicDensityOfStates,
+    XASSpectra,
+    FermiLevel,
+)
+from nomad_simulations.variables import Temperature, Energy2 as Energy
+from nomad_simulations.utils import get_sibling_section
+
+
+class TestSpectralProfile:
+    """
+    Test the `SpectralProfile` class defined in `properties/spectral_profile.py`.
+    """
+
+    def test_is_valid_spectral_profile(self):
+        """
+        Test the `is_valid_spectral_profile` method.
+        """
+        spectral_profile = SpectralProfile(
+            variables=[Energy(grid_points=[-3, -2, -1, 0, 1, 2, 3] * ureg.joule)]
```

Comment:
If you want to avoid conversions and stay with the default, you can leave out `ureg.joule`.

Reply:
```python
grid_points=list(range(-3, 4))
```